### PR TITLE
[LIVY-645]Add Session Name, Owner, Proxy User information to Web UI

### PR DESCRIPTION
**Pull request by yantzu**
## What changes were proposed in this pull request?

1, Web UI, add Session Name to Interactive Sessions list
2, Web UI, add Session Name, Owner, Proxy User to Batch Sessions list
3,~~fix thrift server session doesn't set Session Name issue.~~ Move to PR #218

## How was this patch tested?

Update existing unit test, and have tested manually.

AddName
AddName2

### CODE REVIEW

File: thriftserver/server/src/main/scala/org/apache/livy/thriftserver/LivyThriftSessionManager.scala [hunk 1]
```diff
@@ -228,7 +228,7 @@ class LivyThriftSessionManager(val server: LivyThriftServer, val livyConf: LivyC
       createInteractiveRequest.kind = Spark
       val newSession = InteractiveSession.create(
         server.livySessionManager.nextId(),
-        None,
+        createInteractiveRequest.name,
```
Comment (mgaido91) [inline thriftserver/server/src/main/scala/org/apache/livy/thriftserver/LivyThriftSessionManager.scala]: shall we do this in a separate PR and add UT for it, since this is a bug?

Comment (yantzu) [reply thriftserver/server/src/main/scala/org/apache/livy/thriftserver/LivyThriftSessionManager.scala]: OK, I will create separate PR.

Comment (jerryshao) [reply thriftserver/server/src/main/scala/org/apache/livy/thriftserver/LivyThriftSessionManager.scala]: @yantzu can you please revert the changes here, since you will create another PR to track this issue.

Comment (yantzu) [reply thriftserver/server/src/main/scala/org/apache/livy/thriftserver/LivyThriftSessionManager.scala]: Hi @jerryshao, I have submitted another commit to revert the change.

File: server/src/main/scala/org/apache/livy/server/batch/BatchSessionServlet.scala [hunk 2]
```diff
@@ -73,8 +75,8 @@ class BatchSessionServlet(
       } else {
         Nil
       }
-    BatchSessionView(session.id, session.name, session.state.toString, session.appId,
-      session.appInfo, logs)
+    BatchSessionView(session.id, session.name, session.owner, session.proxyUser,
+      session.state.toString, session.appId, session.appInfo, logs)
```
Comment (jerryshao) [inline server/src/main/scala/org/apache/livy/server/batch/BatchSessionServlet.scala]: I think you can match this with interactive session's view. 1. interactive session uses `null` for empty proxy user; 2. interactive session may have more field, maybe we can also add these in batch session.

Comment (yantzu) [reply server/src/main/scala/org/apache/livy/server/batch/BatchSessionServlet.scala]: Hi @jerryshao , sorry for late. My consideration is:
1, I use Option[String] as proxyUser type is because BatchRecoveryMetadata's proxyUser is Option[String]. And name, appId are Option[String], my purpose it to make them consistent. Interactive session is a little different because its SessionInfo class is Java
2, The only field batch session doesn't have is "kind", which is a legacy field, batch session doesn't have the field information.
